src/helpers/schema-miller.ts

```diff
@@ -59,53 +56,37 @@ export default class SchemaMiller {
 
     // Traverse the schema and collect generated files
     if (node.mill) {
-      const normal = SchemaMiller.normalizeOptions(node.opts || {})
+      const normal = SchemaMiller.normalizeOptions(node)
       const resolved = SchemaMiller.resolveDependency(normal, dir)
-      let headers = {}
       let form
-      if (resolved.opts.use) {
+      if (resolved.use) {
         form = undefined
       } else if (typeof payload === 'function') {
         form = payload()
       } else {
         form = payload
       }
-      if (form && form.getHeaders) {
-        headers = form.getHeaders()
-      } else {
-        headers = {
-          'content-type': 'application/json'
-        }
-      }
-      const file = await remoteMill(node.mill, resolved, form, headers)
+      const file = await remoteMill(node.mill, resolved, form, {})
       dir.files[':single'] = file
     } else if (node.links) {
       // Determine order
       const steps = SchemaMiller.sortLinksByDependency(node.links)
       // Send each link
       // eslint-disable-next-line no-restricted-syntax
       for (const step of steps) {
-        const body = payload
         let form
-        const normal = SchemaMiller.normalizeOptions(step.link.opts || {})
+        const normal = SchemaMiller.normalizeOptions(step.link || {})
         const resolved = SchemaMiller.resolveDependency(normal, dir)
-        let headers = {}
-
-        if (resolved.opts.use) {
-          // It's a file, the hash will pass as the payload, don't send file again
+        if (resolved.use) {
+          // It's an existing 'file', hash will pass as payload, so don't send file again
           form = undefined
-        } else if (typeof body === 'function') {
-          form = body()
+        } else if (typeof payload === 'function') {
+          form = payload()
         } else {
-          form = body
-        }
-
-        if (form && form.getHeaders) {
-          headers = form.getHeaders()
+          form = payload
         }
-
         // Must be synchronous for dependencies
-        const file = await remoteMill(step.link.mill, resolved, form, headers)
+        const file = await remoteMill(step.link.mill, resolved, form, {})
```

Comment:
leaving this option in here just in case, but we could probably remove it from that function